.cursor/rules/code-style.mdc

```diff
@@ -0,0 +1,469 @@
+           self.config: Dict[str, Any] = config
+           self.cache: Dict[str, Any] = {}
+   
+   # ❌ Incorrect
+   class DataProcessor:
+       def __init__(self, config):
+           self.config = config  # No type annotation
+   ```
+
+5. **Use Protocol classes for structural typing**:
+   ```python
+   # ✅ Correct
+   from typing import Protocol
+   
+   class DataHandler(Protocol):
+       def process(self, data: str) -> str: ...
+   
+   def use_handler(handler: DataHandler) -> None:
+       pass
+   ```
+
+6. **Handle Optional types properly**:
+   ```python
+   # ✅ Correct
+   def process_optional(value: Optional[str]) -> str:
+       if value is None:
+           return "default"
+       return value.upper()
+   
+   # ❌ Incorrect - potential None access
+   def process_optional(value: Optional[str]) -> str:
+       return value.upper()  # Could fail if value is None
+   ```
+
+### MyPy Configuration Requirements
+
+- All new code must pass MyPy strict mode
+- Use `# type: ignore` comments only when absolutely necessary and document why
+- Prefer `# type: ignore[error-code]` over blanket ignores
+- Run MyPy locally before committing: `mypy src/`
+
+### Common Typing Patterns
+
+1. **Async functions**:
+   ```python
+   async def fetch_data(url: str) -> Dict[str, Any]:
+       pass
+   ```
+
+2. **Class methods**:
+   ```python
+   class MyClass:
+       def instance_method(self, param: str) -> int:
+           pass
+       
+       @classmethod
+       def class_method(cls, param: str) -> "MyClass":
+           pass
+       
+       @staticmethod
+       def static_method(param: str) -> bool:
+           pass
+   ```
+
+3. **Type aliases for complex types**:
+   ```python
+   from typing import TypeAlias
+   
+   ConfigDict: TypeAlias = Dict[str, Union[str, int, bool]]
+   ResultList: TypeAlias = List[Dict[str, Any]]
+   ```
+
+## OS Operations Must Use Adaptors
+
+**CRITICAL**: Never use OS operations directly in `src/` code. Always use adaptors for better testability and maintainability.
+
+### OS Adaptor Requirements
+
+1. **Use existing adaptors in `src/dd_license_attribution/adaptors/`**:
+   ```python
+   # ✅ Correct
+   from dd_license_attribution.adaptors.os import OSAdaptor
+   
+   class FileProcessor:
+       def __init__(self, os_adaptor: OSAdaptor) -> None:
+           self.os_adaptor = os_adaptor
+       
+       def process_file(self, path: str) -> str:
+           if self.os_adaptor.path_exists(path):
+               return self.os_adaptor.read_file(path)
+           return ""
+   
+   # ❌ Incorrect - direct OS usage
+   import os
+   
+   class FileProcessor:
+       def process_file(self, path: str) -> str:
+           if os.path.exists(path):  # Direct OS usage
+               with open(path) as f:  # Direct OS usage
+                   return f.read()
+           return ""
+   ```
+
+2. **Create new adaptors when needed**:
+   ```python
+   # ✅ Correct - new adaptor
```

Comment:
Currently, we are not using classes for adaptors

Reply:
Yes, but it make sense to group all those methods in a class.